src/bin/sus-kernel/permission/verify/parsed_sudoers_type.rs

```diff
@@ -0,0 +1,159 @@
+// Since sudoers store users by username and not uid,
+// we use special types in this file to easily query sudoers using ids
+use super::sudoers_type;
+use super::sudoers_type::Command;
+use super::sudoers_type::User::{Useralias, Usergroup, Username};
+use super::Permission;
+use super::ALL;
+use nix::unistd::{Gid, Uid};
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::ffi::CString;
+use users::{get_group_by_name, get_user_by_name};
+
+#[derive(Debug)]
+pub struct AllowedCmd {
+    pub users: HashSet<Uid>,
+    pub allow_all_users: bool,
+    pub groups: HashSet<Gid>,
+    pub allow_all_groups: bool,
+    pub paths: HashSet<CString>,
+    pub allow_all_cmds: bool,
+    pub options: Vec<sudoers_type::Option>,
+}
+
+impl AllowedCmd {
+    pub fn new() -> Self {
+        AllowedCmd {
+            users: HashSet::new(),
+            groups: HashSet::new(),
+            paths: HashSet::new(),
+            options: Vec::new(),
+            allow_all_cmds: false,
+            allow_all_users: false,
+            allow_all_groups: false,
+        }
+    }
```

Comment:
Is this constructor ever used? If not, it should probably be removed.

Reply:
yup, it's used when we are generating cmd specs for each rules